src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -158,6 +160,34 @@ internal static bool VerifyRestoreOutput(RestoreRequest request, CacheFile cache
                 return false;
             }
 
+            if (request.ForceUpdatePackageLastAccessTime)
+            {
+                foreach (var package in cacheFile.ExpectedPackageFilePaths)
+                {
+                    var packageFolder = new DirectoryInfo(Directory.GetParent(package).FullName).Parent.Parent.FullName;
```

Comment:
Are we trying to resolve `%USERPROFILE%/.nuget` folder on Windows here? If yes, can we use `NuGetEnvironment.GetFolderPath(NuGetFolderPath.NuGetHome)` instead?

I am curious if customers configure a different [`globalPackagesFolder`](https://docs.microsoft.com/en-us/nuget/Consume-Packages/managing-the-global-packages-and-cache-folders)in nuget.config.

Reply:
> I am curious what happens if customers configure different [globalPackagesFolder](https://docs.microsoft.com/en-us/nuget/Consume-Packages/managing-the-global-packages-and-cache-folders)in nuget.config.

Since this code is just going back in the file system hierarchy, my expectation is that it would just work.

> Maybe it is worth adding a comment to code.

Sure.

> can we use NuGetEnvironment.GetFolderPath(NuGetFolderPath.NuGetHome) instead?

In the scenario you have mentioned, if a user sets a custom [globalPackagesFolder](https://docs.microsoft.com/en-us/nuget/Consume-Packages/managing-the-global-packages-and-cache-folders) in nuget.config, my understanding is that  `NuGetEnvironment.GetFolderPath(NuGetFolderPath.NuGetHome)` won't return the intended location while the current code in the PR would.

Reply:
From a quick debug, I see that the global packages folder path already exists in `request.PackagesDirectory`. 

A whole lot of allocation can be avoided if this, and line 169 below, is replaced with `if (package.StartsWith(request.PackagesDirectory)) { continue; }`.